app/views/posts/_post.html.erb

```diff
@@ -38,8 +38,8 @@
     <% end %>
 
     <% if post.comments.any? && !hide_feedbacks %>
-      <span class="comment-hint text-muted" title="<%= pluralize(post.comments.count, 'comment') %>">
-        <span class="glyphicon glyphicon-comment"></span> &times;<%= post.comments.count %>
+      <span class="comment-hint text-muted" title="<%= pluralize(post.comments.size, 'comment') %>">
```

Comment:
This won't remove the additional query here. Rails overrides `size` to be the same as `count`, so it still does the query. You want to add `.includes(:comments)` to the query in the controller (or add `:comments` to an existing `includes`) , then use the normal `count` here.

Reply:
Unless there are some versioning issues involved, AR's `Relation#size` looks like [this](https://github.com/rails/rails/blob/5f3ff60084ab5d5921ca3499814e4697f8350ee7/activerecord/lib/active_record/relation.rb#L259)
```ruby
    def size
      loaded? ? @records.length : count(:all)
    end
```
which I suppose only issues `COUNT(*)` if the records are not yet loaded?

Reply:
In addition, I believe that `:comments` are already `includes`-ed in `includes_for_post_row`

Reply:
> which I suppose only issues COUNT(*) if the records are not yet loaded?

Correct, so if it's sending a COUNT query then the records aren't loaded yet.

Reply:
Your alternative is to add a single extra query in the controller along these lines:

```ruby
@comment_counts = Comment.where(post_id: @posts.map(&:id)).group(:post_id).count
```

then in the view, instead of just `post.comments.count`:

```erb
defined?(@comment_counts) ? @comment_counts[post.id] : post.comments.count
```

Reply:
> then the records aren't loaded yet.

Pretty sure we have a misunderstanding here.

1. `Relation#count` always use a database `COUNT(*)` query regardless if the records are already loaded.
2. `Relation#size` is smarter and only uses `COUNT(*)` if the records aren't loaded yet
3. We have **already** been loading `:comments` in `Post.all.includes_for_post_row`, so changing `count` to `size` solves N+1 query problem here.

Reply:
Ah, I see where you're going. Have you checked either in Rails log or in miniprofiler that `size` does actually stop the `COUNT` queries being run?

Reply:
Those queries no longer show up in foreman logs.